proc/test_thread_rand.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-tests: random lifespan threads test
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	unsigned int time;
+	unsigned int id;
+} thread_params_t;
+
+
+typedef struct {
+	thread_params_t params;
+	char stack[512] __attribute__((aligned(8))); /* stack needs to be aligned to 8 */
+} thread_self_t;
+
+
+typedef struct {
+	unsigned int fin;
+	handle_t fin_lock;
+	thread_self_t threads[10]; /* set threads amount here */
+} threads_common_t;
+
+
+static threads_common_t threads_common;
+
+
+static void test_thread(void *arg)
+{
+	thread_params_t *params = (thread_params_t *)arg;
+	int err;
+	usleep(params->time * 1000);
+
+	err = mutexLock(threads_common.fin_lock);
+	TEST_ASSERT_EQUAL_INT(err, EOK);
```

Comment:
please note that the `TEST_ASSERT_EQUAL_INT` is defined as: `TEST_ASSERT_EQUAL_INT(expected, actual)`, so You need to change the order of params for some of the calls).

Reply:
Checked and changed if necessary. Thanks.

Reply:
I think we should rather not use `TEST_ASSERT*` in threads. It uses `longjmp` and/or sets a global state which is not protected by any synchronization primitives.

Reply:
Ok, so the reasonable option is to in some way escape the error from thread. I can make another parameter in `thread_params_t` for that.

Reply:
@nalajcie , is this an issue?

Reply:
Oh, sorry, didn't know that it uses `longjmp`. As doing `longjmp` from different thread that `setjmp` is undefined behavior - it may (and probably will) result in kernel panic or some strange errors (You may test it by Yourself ny failing on purpose).

For now we can add some result code in `thread_params_t` and verify it in the main thread, maybe we would come up with better solution for multithreading support in `unity` framework in the future.

Reply:
Thanks! Going on with new field in `thread_params_t` then.